File: cpp/szyfr_cezara.cpp
/*
 * szyfr_cezara.cpp
 */


#include <iostream>
using namespace std;

void male_litery(char t[]){
    int i = 0;
    while (t[i] != '\0'){
        //for (i = 65; i<91; i++){
        //cout << tabzn[i] << i+32 <<(int)tabzn[i] << endl;
        if ((int)t[i] > 64 || (int)t[i] < 90) {
            t[i] = (char)((int)t[i]+32);
        }
        i++;
    }
}

void szyfruj(char t[], int k) {
    int i = 0;
    int kod = 0;
    while (t[i] != '\0'{
        kod = (int)t[i] + k;
        if (kod > 122) {
            kod -= 26;
        }
        t[i] = (char)kod;
        i++;
    }
}

void deszyfruj(char t[], int k) {
    int i = 0;
    int kod = 0;
    while (t[i] != '\0'{
        kod = (int)t[i] - k;
        t[i] = (char)kod;
        i++;
    }
}

int main(int argc, char **argv)
{
    int r = 30;
    char tekst[r];
    cout << "podaj tekst; " << endl;
    cin.getline(tekst, r); 
    int klucz = 0
    cout << "podaj klucz: " << endl;
    cin >> klucz;
    klucz = klucz % 26;
    szyfruj(tekst, klucz);
    cout << tekst << endl;
    deszyfruj(tekst, klucz);
    cout << tekst << endl;
       
	return 0;
}

File: python/szyfr_cezara.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  szyfr_cezara.py

def pobierz_klucz():
    try:
        klucz = int(input('podaj klucz:\n'))
        return klucz % 26
    except ValueError:
        print('bledny klucz!')
        return 3   
        
def szyfruj_1(tekst, klucz):
    szyfrogram = ''
    for znak in tekst:
        ascii = ord(znak)
        if ascii + klucz > 122:
            znak = chr(ascii + klucz - 26)
        else:
            znak = chr(ascii + klucz)
        szyfrogram += znak
    print('Szyfrogram:\n', szyfrogram)
        
def main(args):
    klucz = input('podaj tekst:\n')
    klucz = pobierz_klucz()
    szyfruj_1(tekst, klucz)
    return 0
    

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
